p4-16/spec/P4-16-spec.adoc

```diff
@@ -3841,18 +3841,23 @@ Additionally, the maximum size of a variable-length bit-string can be determined
 [#sec-casts]
 === Casts
 
-P4 provides a limited set of casts between types. A cast is written
-`(t) e`, where `t` is a type and `e` is an expression. Casts are only
-permitted on base types and derived types introduced by `typedef`, `type`, and `enum`.
-While this design is arguably more onerous for programmers, it has several benefits:
+P4 provides a set of casts between types. A cast is written
+`(t) e`, where `t` is a type and `e` is an expression. Casts in P4 are
+permitted only under the conditions specified in section <<sec-explicit-casts>>
+for explicit casts and section <<sec-implicit-casts>> for implicit casts.
+By allowing casts only in limited scenarios, this restriction is more
+onerous for programmers, but it also provides several benefits.
```

Comment:
nitpick: maybe this would read more naturally:

```suggestion
Allowing casts only in limited scenarios is more onerous for
programmers, but it also provides several benefits.